x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/MatchPhrase.java

```diff
@@ -0,0 +1,345 @@
+import static org.elasticsearch.xpack.esql.expression.predicate.operator.comparison.EsqlBinaryComparison.formatIncompatibleTypesMessage;
+
+/**
+ * Full text function that performs a {@link org.elasticsearch.xpack.esql.querydsl.query.MatchPhraseQuery} .
+ */
+public class MatchPhrase extends FullTextFunction implements OptionalArgument, PostAnalysisPlanVerificationAware {
+
+    public static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(
+        Expression.class,
+        "MatchPhrase",
+        MatchPhrase::readFrom
+    );
+    public static final Set<DataType> FIELD_DATA_TYPES = Set.of(KEYWORD, TEXT);
+    public static final Set<DataType> QUERY_DATA_TYPES = Set.of(KEYWORD, TEXT);
+
+    protected final Expression field;
+
+    // Options for match_phrase function. They don’t need to be serialized as the data nodes will retrieve them from the query builder
+    private final transient Expression options;
+
+    public static final Map<String, DataType> ALLOWED_OPTIONS = Map.ofEntries(
+        entry(ANALYZER_FIELD.getPreferredName(), KEYWORD),
+        entry(BOOST_FIELD.getPreferredName(), FLOAT),
+        entry(SLOP_FIELD.getPreferredName(), INTEGER),
+        entry(ZERO_TERMS_QUERY_FIELD.getPreferredName(), KEYWORD)
+    );
+
+    @FunctionInfo(
+        returnType = "boolean",
+        preview = true,
+        description = """
+            Use `MATCH_PHRASE` to perform a <<query-dsl-match-query-phrase,match_phrase query>> on the specified field.
+            Using `MATCH_PHRASE` is equivalent to using the `match_phrase` query in the Elasticsearch Query DSL.
+
+            MatchPhrase can be used on <<text, text>> fields, as well as other field types like keyword, boolean, or date types.
+            MatchPhrase is not supported for <<semantic-text, semantic_text>> or numeric types.
+
+            MatchPhrase can use <<esql-function-named-params,function named parameters>> to specify additional options for the
+            match_phrase query.
+            All [`match_phrase`](/reference/query-languages/query-dsl/query-dsl-match-query-phrase.md) query parameters are supported.
+
+            `MATCH_PHRASE` returns true if the provided query matches the row.""",
+        examples = {
+            @Example(file = "match-phrase-function", tag = "match-phrase-with-field"),
+            @Example(file = "match-phrase-function", tag = "match-phrase-with-named-function-params") },
+        appliesTo = {
+            @FunctionAppliesTo(
+                lifeCycle = FunctionAppliesToLifecycle.COMING,
+                description = "Support for optional named parameters is only available in serverless, or in a future {{es}} release"
+            ) }
+    )
+    public MatchPhrase(
+        Source source,
+        @Param(name = "field", type = { "keyword", "text" }, description = "Field that the query will target.") Expression field,
+        @Param(
+            name = "query",
+            type = { "keyword", "text" },
```

Comment:
Nit - Query can't really be a text field

```suggestion
            type = { "keyword" },
```